Running rnx-bundle fails after updating to latest rnx-kit/dep-check and rnx-kit/metro-config

### What happened?

If I revert the latest bumps to both packages, the command runs as expected.
Running react-native rnx-bundle gives the following errors:

yarn bundle
yarn run v1.22.19
warning package.json: No license field
$ react-native rnx-bundle --dev false
warn Package @rnx-kit/cli has been ignored because it contains invalid configuration. Reason: Cannot find module 'node:path'
Require stack:
- E:\fluentui-react-native\node_modules\@rnx-kit\dep-check\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\lib\dep-check.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\react-native.config.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\import-fresh\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\loaders.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\createExplorer.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\tools\config\readConfigFromDisk.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\tools\config\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\commands\install\install.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\commands\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\index.js
- E:\fluentui-react-native\node_modules\react-native\cli.js
error Unrecognized command "rnx-bundle". Did you mean "ram-bundle"?
info Run "react-native --help" to see a list of all available commands.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


yarn bundle
yarn run v1.22.19
warning package.json: No license field
$ react-native rnx-bundle --dev false
error Cannot find module 'node:path'
Require stack:
- E:\fluentui-react-native\node_modules\@rnx-kit\tools-workspaces\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\metro-config\src\index.js
- E:\fluentui-react-native\apps\win32\metro.config.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\import-fresh\index.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\cosmiconfig\dist\loaders.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\cosmiconfig\dist\createExplorer.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\cosmiconfig\dist\index.js
- E:\fluentui-react-native\node_modules\metro-config\src\loadConfig.js
- E:\fluentui-react-native\node_modules\metro-config\src\index.js
- E:\fluentui-react-native\node_modules\metro\src\commands\build.js
- E:\fluentui-react-native\node_modules\metro\src\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\metro-service\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\lib\bundle.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\react-native.config.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\import-fresh\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\loaders.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\createExplorer.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\tools\config\readConfigFromDisk.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\tools\config\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\commands\install\install.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\commands\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\index.js
- E:\fluentui-react-native\node_modules\react-native\cli.js.
Error: Cannot find module 'node:path'
Require stack:
- E:\fluentui-react-native\node_modules\@rnx-kit\tools-workspaces\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\metro-config\src\index.js
- E:\fluentui-react-native\apps\win32\metro.config.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\import-fresh\index.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\cosmiconfig\dist\loaders.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\cosmiconfig\dist\createExplorer.js
- E:\fluentui-react-native\node_modules\metro-config\node_modules\cosmiconfig\dist\index.js
- E:\fluentui-react-native\node_modules\metro-config\src\loadConfig.js
- E:\fluentui-react-native\node_modules\metro-config\src\index.js
- E:\fluentui-react-native\node_modules\metro\src\commands\build.js
- E:\fluentui-react-native\node_modules\metro\src\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\metro-service\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\lib\bundle.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\lib\index.js
- E:\fluentui-react-native\node_modules\@rnx-kit\cli\react-native.config.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\import-fresh\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\loaders.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\createExplorer.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\node_modules\cosmiconfig\dist\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\tools\config\readConfigFromDisk.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\tools\config\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\commands\install\install.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\commands\index.js
- E:\fluentui-react-native\node_modules\@react-native-community\cli\build\index.js
- E:\fluentui-react-native\node_modules\react-native\cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (E:\fluentui-react-native\node_modules\@rnx-kit\tools-workspaces\lib\index.js:36:27)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Module.require (internal/modules/cjs/loader.js:974:19)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Seems like the new workspace-tools package that was added (https://github.com/microsoft/rnx-kit/commit/37245c33c63f3e676121167ab248b72522cdc88e) imports from the referenced node:path package and it's failing to find it on our machines? Interestingly our CI machine runs fine.

My installed Node version is v14.17.1 which I think is within the declared support range. But I read somewhere that node:path may only be supported on >= v16?

### Affected Package

@rnx-kit/dep-check and @rnx-kit/metro-config

### Version

1.12.18 and 1.2.34, respectively

### Which platforms are you seeing this issue on?

- [ ] Android
- [ ] iOS
- [ ] macOS
- [X] Windows

### System Information

```true
System:
    OS: Windows 10 10.0.22000
    CPU: (32) x64 AMD Ryzen Threadripper PRO 3955WX 16-Cores
    Memory: 111.58 GB / 127.86 GB
  Binaries:
    Node: 14.17.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7621141
    Visual Studio: 17.2.32602.215 (Visual Studio Enterprise 2022), 16.11.32602.291 (Visual Studio Enterprise 2019)
  Languages:
    Java: 11.0.12 - C:\Program Files\Microsoft\jdk-11.0.12.7-hotspot\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```


### Steps to Reproduce

1. Pull latest https://github.com/microsoft/fluentui-react-native
2. yarn
3. yarn build
4. cd apps\win32
5. yarn bundle

May need to change node to v 14.17.1 or similar version

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/.changeset/warm-balloons-remain.md
+++ b/.changeset/warm-balloons-remain.md
@@ -1,0 +1,5 @@
+---
+"@rnx-kit/tools-workspaces": patch
+---
+
+Remove `node:` prefix to support Node <14.18

--- a/incubator/build/package.json
+++ b/incubator/build/package.json
@@ -46,7 +46,7 @@
     "memfs": "^3.4.1"
   },
   "engines": {
-    "node": ">=14.15"
+    "node": ">=14.18"
   },
   "eslintConfig": {
     "extends": "@rnx-kit/eslint-config"

--- a/incubator/patcher-rnmacos/package.json
+++ b/incubator/patcher-rnmacos/package.json
@@ -21,7 +21,7 @@
     "directory": "incubator/patcher-rnmacos"
   },
   "engines": {
-    "node": ">= 14"
+    "node": ">=14"
   },
   "scripts": {
     "prepare": "echo '⚠️ patcher-rnmacos is EXPERIMENTAL - USE WITH CAUTION  ⚠️'",

--- a/incubator/rn-changelog-generator/package.json
+++ b/incubator/rn-changelog-generator/package.json
@@ -18,7 +18,7 @@
     "directory": "incubator/rn-changelog-generator"
   },
   "engines": {
-    "node": ">= 14"
+    "node": ">=14"
   },
   "scripts": {
     "prepare": "echo '⚠️ rn-changelog-generator is EXPERIMENTAL - USE WITH CAUTION  ⚠️'",

--- a/package.json
+++ b/package.json
@@ -48,7 +48,7 @@
     "prettier": "^2.3.0"
   },
   "engines": {
-    "node": ">=14.15"
+    "node": ">=14.18"
   },
   "resolutions": {
     "eslint-plugin-react": "^7.26.0",

--- a/packages/golang/package.json
+++ b/packages/golang/package.json
@@ -15,7 +15,7 @@
     "directory": "packages/golang"
   },
   "engines": {
-    "node": ">= 14"
+    "node": ">=14"
   },
   "scripts": {
     "build": "tsc",

--- a/packages/metro-service/package.json
+++ b/packages/metro-service/package.json
@@ -21,7 +21,7 @@
     "test": "rnx-kit-scripts test"
   },
   "engines": {
-    "node": ">= 10.12"
+    "node": ">=10.12"
   },
   "dependencies": {
     "@rnx-kit/tools-language": "^1.2.6",

--- a/packages/template/package.json
+++ b/packages/template/package.json
@@ -20,7 +20,7 @@
     "directory": "packages/template"
   },
   "engines": {
-    "node": ">= 14"
+    "node": ">=14.18"
   },
   "scripts": {
     "build": "rnx-kit-scripts build",

--- a/packages/tools-workspaces/package.json
+++ b/packages/tools-workspaces/package.json
@@ -34,7 +34,7 @@
     "@rnx-kit/scripts": "*"
   },
   "engines": {
-    "node": ">= 14"
+    "node": ">=14.15"
   },
   "eslintConfig": {
     "extends": "@rnx-kit/eslint-config"

--- a/packages/tools-workspaces/src/common.ts
+++ b/packages/tools-workspaces/src/common.ts
@@ -1,7 +1,7 @@
 import type { Options } from "fast-glob";
 import fg from "fast-glob";
 import findUp from "find-up";
-import * as path from "node:path";
+import * as path from "path";
 
 type PackageManager = {
   findWorkspacePackages: (sentinel: string) => Promise<string[]>;

--- a/packages/tools-workspaces/src/index.ts
+++ b/packages/tools-workspaces/src/index.ts
@@ -1,4 +1,4 @@
-import * as path from "node:path";
+import * as path from "path";
 import {
   findSentinel,
   findSentinelSync,

--- a/packages/tools-workspaces/src/lerna.ts
+++ b/packages/tools-workspaces/src/lerna.ts
@@ -1,6 +1,6 @@
-import { existsSync as fileExists, readFileSync } from "node:fs";
-import { readFile } from "node:fs/promises";
-import * as path from "node:path";
+import { existsSync as fileExists, readFileSync } from "fs";
+import { readFile } from "fs/promises";
+import * as path from "path";
 import {
   findPackages,
   findPackagesSync,

--- a/packages/tools-workspaces/src/pnpm.ts
+++ b/packages/tools-workspaces/src/pnpm.ts
@@ -1,4 +1,4 @@
-import * as path from "node:path";
+import * as path from "path";
 import {
   default as readYamlFile,
   sync as readYamlFileSync,

--- a/packages/tools-workspaces/src/rush.ts
+++ b/packages/tools-workspaces/src/rush.ts
@@ -1,6 +1,6 @@
-import { readFileSync } from "node:fs";
-import { readFile } from "node:fs/promises";
-import * as path from "node:path";
+import { readFileSync } from "fs";
+import { readFile } from "fs/promises";
+import * as path from "path";
 
 type RushProject = {
   packageName: string;

--- a/packages/tools-workspaces/src/yarn.ts
+++ b/packages/tools-workspaces/src/yarn.ts
@@ -1,6 +1,6 @@
-import { readFileSync } from "node:fs";
-import { readFile } from "node:fs/promises";
-import * as path from "node:path";
+import { readFileSync } from "fs";
+import { readFile } from "fs/promises";
+import * as path from "path";
 import { findPackages, findPackagesSync } from "./common";
 
 type Manifest = {

